src/main/kotlin/io/hemin/wien/model/BuilderFactory.kt

```diff
@@ -0,0 +1,8 @@
+package io.hemin.wien.model
+
+import io.hemin.wien.builder.Builder
+
+internal interface BuilderFactory<D, out B : Builder<D>> {
+    /** Returns a builder implementation [B] for [D]. */
+    fun builder(): B
+}
```

Comment:
Technically not really necessary, but it helps to avoid stupid implementation errors. Would have been nice if the `@JvmStatic` annotation could have been added here though.